File: Irrgarten_Ruby/Weapon.rb
#encoding:utf-8

module Irrgarten 
	class Weapon
		
		def initialize (power, uses) #float, int
			@power = power
			@uses = uses	
		end
		
		def attack ()
			salida = 0.0
			if @uses > 0 then 
				salida = @power
				@uses -= 1
			end
			salida
		end
		
		def to_s ()
			"W[#{@power},#{@uses}]" 
		end
		
		def discard 
			return Dice.discard_element(@uses)
		end
	end	
end

File: Irrgarten_Java/Irrgarten_P5/src/irrgarten/Weapon.java

package irrgarten;

/**
 *
 * @author leonin04
 */
public class Weapon extends CombatElement{
    
    public Weapon (float power, int uses) {
        super(power,uses);
    }
    
    public float attack () {
        return super.produceEffect();
    }
    
    
    @Override
    public String toString () { 
        String temp = super.toString();
        return "W" + temp;
    }
    
}
